File: UVMonitor/Views/Components/UVIndexCard.swift
import SwiftUI

struct UVIndexCard: View {
    let uvIndex: Double
    let level: UVLevel
    let lastUpdated: Date?

    var body: some View {
        VStack(spacing: 8) {
            Text("UV Index")
                .font(.subheadline)
                .foregroundStyle(.white.opacity(0.8))

            Text(String(format: "%.1f", uvIndex))
                .font(.system(size: 64, weight: .bold, design: .rounded))
                .foregroundStyle(.white)

            Text(level.rawValue.uppercased())
                .font(.headline)
                .fontWeight(.semibold)
                .foregroundStyle(.white)

            if let lastUpdated {
                Text("Updated \(lastUpdated, style: .time)")
                    .font(.caption)
                    .foregroundStyle(.white.opacity(0.7))
            }
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 24)
        .background(
            RoundedRectangle(cornerRadius: 16)
                .fill(level.color.gradient)
        )
    }
}

File: android/app/src/main/java/com/uvmonitor/app/ui/components/UVIndexCard.kt
package com.uvmonitor.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.uvmonitor.app.model.UVLevel
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

@Composable
fun UVIndexCard(
    uvIndex: Double?,
    lastUpdated: Date?,
    modifier: Modifier = Modifier
) {
    val level = UVLevel.fromUVIndex(uvIndex ?: 0.0)
    val gradient = Brush.verticalGradient(
        colors = listOf(
            level.color,
            level.color.copy(alpha = 0.7f)
        )
    )

    Box(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(16.dp))
            .background(gradient)
            .padding(32.dp),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            if (uvIndex != null) {
                Text(
                    text = String.format(Locale.US, "%.1f", uvIndex),
                    fontSize = 64.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White
                )
                Text(
                    text = level.label.uppercase(),
                    fontSize = 18.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.White.copy(alpha = 0.9f)
                )
                if (lastUpdated != null) {
                    val timeFormat = SimpleDateFormat("h:mm a", Locale.getDefault())
                    Text(
                        text = "Updated ${timeFormat.format(lastUpdated)}",
                        fontSize = 13.sp,
                        color = Color.White.copy(alpha = 0.7f)
                    )
                }
            } else {
                Text(
                    text = "—",
                    fontSize = 64.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White
                )
                Text(
                    text = "LOADING",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.White.copy(alpha = 0.9f)
                )
            }
        }
    }
}
